### DISCUSSION TITLE
How much space does 2nd temporary directory need per k32 plot

### DISCUSSION THREAD

**Discussion by nomadengineer** [Chia-Network/chia-blockchain#4299] [Q&A]
Hello,
I need to know the max amount of space needed in the 2nd temporary directory for a k32 plot.

Trying to figure it out for some plotting optimization.

thanks!

**Comment by knightblackm** [comment]
for temp you need 239Gb in temp one, temp 2 didin't use in phase 2.

**Reply by nomadengineer** [reply]
temp 2 is used in phase 3 & 4. so I am trying to determine whats the maximum they are used per plot!

**Reply by Ron-ski** [reply]
I'd like to know as well.

**Reply by room101-dev** [reply]
You have two NVME's, one is 1TB ( -t), and the other is 512MB (-2)
You want to make forty plots, in parallel in 4 threads; so each thread makes 10 plots
The cmd-line invocation is ...

"Chia plot create -t /dev/nvme1/tmp1 -2 /dev/nvme2/tmp -d /dev/sda1/hdd-plots -n 10"
wait 1h;
"Chia plot create -t /dev/nvme1/tmp2 -2 /dev/nvme2/tmp -d /dev/sda1/hdd-plots -n 10"
wait 1h;
"Chia plot create -t /dev/nvme1/tmp3 -2 /dev/nvme2/tmp -d /dev/sda1/hdd-plots -n 10"
wait 1h;
"Chia plot create -t /dev/nvme1/tmp4 -2 /dev/nvme2 -d /dev/sda1/hdd-plots -n 10"

You plot parallel, 4x plots at a time, by staggering the the start of each plot invocation above, but +30m.

Essentially -t is where the work is done, in stage the 3 the plot is compressed to 108gb, and written to -2 ( a fast ram disk ), in final stage the plot is written to -d ( the permanent hdd )

It is up to you to make sure that the -d target can hold 40 plots 40*108gb, it is up to you that the -2 can hold 4 intermediate plots (4 * 108gb)

-t is is tricky, the plot phase is each /tmpN folder can go as high has 280gb, but also as low as 108gb, so by staggering you will always enough space

your question is about -2, its always going to be no larger than 4 plots, now they're not all going to be created at the same time because of 'staggering', on the other hand the worst case is 4 * 108gb, I would not mess around, I use 1TB NVME for both -t & -2, I use 8/16tb for hdd. The 'chia create plot..." will take care of all deletion of temp files, you don't have to do anything, your only job is to make sure that -d doesn't run out of target space.
The way you choose -n x, how many plots on each thread is that if the target drive has say 10tb, then you know it can handle 95 plots, so -n 10 is fine, I would NOT do too much cuz it would be a long time, if you use NVME drives expect 6-8 hours per batch of 4 plots, expect 16 per day, I would never try to parallel more the 4x at a time per NVME, you will destroy the controller.

There are lots of other switches like -r, -b but the defaults are fine. You can of course use 'chia plot create -help' to get a list of all options.

Some will disagree on the above, because they use the GUI, but the GUI still uses these cmd-lines, and the chia-dev gui is shit, I always use the cmd-line so I get exactly what I want, if the gui gets out of sync all is screwed, here by using 'HTOP', you can control each process.
The only other thing to know is to run "SUDO fstrim -v /dev/nvme1" every six hours, this cleans up the NVME drives and tells the controller which blocks are free on the drive, keeps the speed of plotting high, if you fail to do this, the plotting process will grind to a halt over time.

Lastly, make sure you 'format' linux using ZFS, 'discard'; just google if you don't understand, same for fstrim

Typically -2 will only create one plot at a time, but you can't take that chance, if a process is suspended and restarted you could later end up with all plots written at the same time, and if you don't have the capacity you lose 8 hours of work.

Start with one plot, given this question. Monitor "SUDO IOTOP", and "HTOP" closely, stats for NVME are in "SUDO NVME smart-logs /dev/nvme1", pay close attention to busy-controller, temperature, and TBW writes, I find that MY NVME die at about 25% of TBW warranty. It's ok a dead NVME on -t, can still be used as a -2, this is because CHIA-POS kills the random-controller, but not the sequential.
